### PR TITLE
feat: added generic disabled tooltip to filters.

### PR DESCRIPTION
**Pull request by MillenniumFalconMechanic**
### Reviewers
**Functional:** 
@seve 

---

## Changes
- Add generic tooltip to disabled filters indicating filter has no values matching current filter criteria.

### CODE REVIEW

File: frontend/src/components/common/Filter/index.tsx [hunk 1]
```diff
@@ -241,21 +236,6 @@ function filterCategoryValuesWithCount(
   return values.filter(({ count }) => count > 0);
 }
 
-/**
- * Returns true if category is not applicable.
- * @param categoryView
- * @returns true when category  is not applicable.
- */
-function isCategoryNA(categoryView: CategoryView): boolean {
```
Comment (MillenniumFalconMechanic) [inline frontend/src/components/common/Filter/index.tsx:249]: I moved the disabled checks from the filter component to the `useCategoryFilter` hook.

**Review by seve — APPROVED** [review]

**Review by seve — CHANGES_REQUESTED** [review]
oops

File: frontend/src/common/hooks/useCategoryFilter.ts [hunk 2]
```diff
@@ -821,14 +821,14 @@ function buildRangeCategoryView(
   rangeCategory: RangeCategory
 ): RangeCategoryView {
   // Build view model of range category.
-  const rangeView = {
+  const rangeView: RangeCategoryView = {
     key: categoryKey as CategoryKey,
```
Comment (seve) [inline frontend/src/common/hooks/useCategoryFilter.ts:825]: Don't we already have `categoryKey` typed as a result of the arg typing?

**Review by seve — APPROVED** [review]

File: frontend/src/common/hooks/useCategoryFilter.ts (hunk 2 above)
Comment (MillenniumFalconMechanic) [reply frontend/src/common/hooks/useCategoryFilter.ts:825]: Good catch! I'll update here as well as the corresponding areas for select and ontology category views.

**Pull request merged**